cpp/benchmarks/fixture/nvbench_fixture.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/mr/device/arena_memory_resource.hpp>
+#include <rmm/mr/device/cuda_async_memory_resource.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/managed_memory_resource.hpp>
+#include <rmm/mr/device/owning_wrapper.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+#include <string>
+
+namespace cudf {
+namespace detail {
+static std::string rmm_mode_parm{"--rmm_mode"};  ///< RMM mode command-line parameter name
+}  // namespace detail
+
+/**
+ * Base fixture for cudf benchmarks using nvbench.
+ *
+ * Initializes the default memory resource to use the RMM pool device resource.
+ */
+struct nvbench_base_fixture {
+  inline auto make_cuda() { return std::make_shared<rmm::mr::cuda_memory_resource>(); }
+
+  inline auto make_pool()
+  {
+    return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda());
+  }
+
+  inline auto make_async() { return std::make_shared<rmm::mr::cuda_async_memory_resource>(); }
+
+  inline auto make_managed() { return std::make_shared<rmm::mr::managed_memory_resource>(); }
```

Comment:
I think @GregoryKimball may have an interest in supporting "managed pool" as well?

Reply:
@bdice 
Would you have a snippet on how to create a managed-pool memory resource in this context?

Reply:
> I think @GregoryKimball may have an interest in supporting "managed pool" as well?

Yes - very great interest!

Reply:
@davidwendt Try this:


```cpp
  inline auto make_managed_pool()
  {
    return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_managed());
  }
```

The pool resource has a template parameter for its upstream. I think it’ll deduce the right upstream template type.

https://github.com/rapidsai/rmm/blob/005992676c9412b724278a91555536635698e3aa/include/rmm/mr/device/pool_memory_resource.hpp#L60-L61

Reply:
Resolved in 0d1a2de.